src/main/resources/io/jenkins/plugins/artifact_manager_s3/S3BlobStoreConfig/help-prefix.html

```diff
@@ -0,0 +1 @@
+<div>This is the string that will be used as prefix for every file path, it must end with an "/" if it is a folder.</div>
```

Comment:
So, missing `public FormValidation doCheckPrefix(@QueryParameter String prefix)`.

Reply:
no, it is waiting in other PR, once we have this one merged I will add the validations, one step a time.

Reply:
Makes things harder to review when they are artificially split up like this. The natural review is of a PR which defines a field, displays its help, _and_ enforces its format via form validation.

Reply:
I try it it is really flaky, most of the times it breaks by timeout getting the config form

```
    @Test
    public void checkConfigurationForm(){
        rr.then(r -> {
            S3BlobStoreConfig s3BlobStoreConfig = S3BlobStoreConfig.get();

            assertNull("container not set initially", s3BlobStoreConfig.getContainer());
            assertNull("prefix not set initially", s3BlobStoreConfig.getPrefix());
            assertNull("region not set initially", s3BlobStoreConfig.getRegion());

            HtmlPage configurePage = r.createWebClient().goTo("configure");
            HtmlForm configForm = configurePage.getFormByName("config");
            setFormInputValue(configForm, "_.container", CONTAINER_NAME);
            setFormInputValue(configForm, "_.prefix", CONTAINER_PREFIX);
            setFormSelectValue(configForm, "_.region", CONTAINER_REGION);
            r.submit(configForm);

            s3BlobStoreConfig = S3BlobStoreConfig.get();
            assertEquals("global config page let us edit container", CONTAINER_NAME, s3BlobStoreConfig.getContainer());
            assertEquals("global config page let us edit prefix", CONTAINER_PREFIX, s3BlobStoreConfig.getPrefix());
            assertEquals("global config page let us edit region", CONTAINER_REGION, s3BlobStoreConfig.getRegion());
        });
        rr.then(r -> {
            S3BlobStoreConfig s3BlobStoreConfig = S3BlobStoreConfig.get();
            assertEquals("global config page let us edit container", CONTAINER_NAME, s3BlobStoreConfig.getContainer());
            assertEquals("global config page let us edit prefix", CONTAINER_PREFIX, s3BlobStoreConfig.getPrefix());
            assertEquals("global config page let us edit region", CONTAINER_REGION, s3BlobStoreConfig.getRegion());
        });
    }

    private void setFormInputValue(HtmlForm config, String fieldName, String value){
        HtmlTextInput textbox = config.getInputByName(fieldName);
        textbox.setText(value);
    }

    private void setFormSelectValue(HtmlForm config, String fieldName, String value){
        HtmlSelect textbox = config.getSelectByName(fieldName);
        textbox.setSelectedAttribute(value, true);
    }
```

 so I made the simple version validating the fields again

```
    @Test
    public void checkConfigurationManually() throws Exception {
        S3BlobStore provider = new S3BlobStore();
        S3BlobStoreConfig s3BlobStoreConfig = S3BlobStoreConfig.get();
        s3BlobStoreConfig.setContainer(CONTAINER_NAME);
        s3BlobStoreConfig.setPrefix(CONTAINER_PREFIX);
        s3BlobStoreConfig.setRegion(CONTAINER_REGION);

        JCloudsArtifactManagerFactory artifactManagerFactory = new JCloudsArtifactManagerFactory(provider);
        ArtifactManagerConfiguration.get().getArtifactManagerFactories().add(artifactManagerFactory);

        LOGGER.info(artifactManagerFactory.getProvider().toString());

        checkFieldValues(artifactManagerFactory);

        //check configuration page submit
        j.configRoundtrip();
        checkFieldValues(artifactManagerFactory);
    }

    private void checkFieldValues(JCloudsArtifactManagerFactory artifactManagerFactory) {
        assertEquals(artifactManagerFactory.getProvider().getContainer(), CONTAINER_NAME);
        assertEquals(artifactManagerFactory.getProvider().getPrefix(), CONTAINER_PREFIX);
        assertTrue(artifactManagerFactory.getProvider() instanceof S3BlobStore);
        assertEquals(((S3BlobStore)artifactManagerFactory.getProvider()).getRegion(), CONTAINER_REGION);
    }
```

Reply:
Timeout?

Reply:
yep, in this line `HtmlPage configurePage = r.createWebClient().goTo("configure");` 40% of the times, too flaky

Reply:
Strange, I do not recall recurrent timeouts in any other HtmlUnit-based tests doing this kind of thing. Anything notable in the thread dumps?

Reply:
It is funny, I just run it ten times to grab the dumps and it does not fail, maybe my laptop didn't want to work yesterday 🤷‍♂️

Reply:
Of course…